.github/workflows/wheels.yml

```diff
@@ -0,0 +1,119 @@
+name: Python wheels
+
+on:
+  workflow_dispatch:
```

Comment:
I don't like this being a dispatch only. Better to have the logic as an action and run it with dispatch when needed and with PRs

Reply:
Since it's not using sscache (for now?), the jobs are very slow (up to 2 hours on mac), I'd rather keep them workflow only and we can improve it later.

Reply:
Using sccache should be trivial.